extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReader.java

```diff
@@ -52,4 +53,20 @@ default long numRows() throws IOException {
      */
     @NotNull
     Dictionary getDictionary();
+
+    /**
+     * Set the channel context to use while reading the parquet file. This will cache a copy of the object for use
+     * across reads. Therefore, its is recommended to use {@link #clearChannelContext()} when done with the
+     * reader/context.
+     *
+     * @param context The channel context to use.
+     */
+    void setChannelContext(@NotNull final SeekableChannelsProvider.ChannelContext context);
```

Comment:
Do we really need this?
We have a separate (probably bad) pattern of passing a context into the constructor. I prefer this (set + clear) to the constructor version, but maybe we just want to pass contexts down rather than do this? It's "safe" as long as accesses are always synchronized/single-threaded, but feels like an anti-pattern.